### ISSUE TITLE
KafkaProducers aren't properly closed

### ISSUE THREAD

**Issue by emenaceb** [spring-cloud/spring-cloud-stream-binder-kafka#1034]
KafkaProducers created by kafka binder aren't closed.
Producer's KafkaThreads aren't stopped

These producers doesn't live in Spring context because they are created by `DefaultKafkaProducerFactory `that are created by `KafkaMessageChannelBinder` and lives inside a `ProducerConfigurationMessageHandler `(subclass of `KafkaProducerMessageHandler`) that is created ny the binder, so they are never published as a bean in any `ApplicationContext`.

They'll **never** listen to application context close events

On the other hand `ProducerConfigurationMessageHandler `implements `Lifecycle `and its `stop()` method is called when Spring does the unbinding.

This method in the version that I'm using (3.0.10) and  in the latest version is like that:

```java
@Override
public void stop() {
	if (this.producerFactory instanceof Lifecycle) {
		((Lifecycle) producerFactory).stop();
	}
	this.running = false;
}
```

But no subclass of `ProducerFactory` implements `Lifecycle`.

Instead of that, all of them (including `DefaultKafkaProducerFactory`) implements `DiposableBean`, but nobody is invoking
`dispose()` method, so threads started by `DefaultKafkaProducerFactory` are **never stopped** (and may be, pending messages are not flushed to the broker).

Note: May be it happens also in kafka streams binder, but I haven't checked.

**Comment by sobychacko** [comment]
@emenaceb Thank you for spotting this. It seems to be a bug, we will address.

**Comment by emenaceb** [comment]
You're welcome 
:-)

**Comment by emenaceb** [comment]
Is It possible to have this fixed in a 3.0.x version?
I'm not sure about compatibility of the binder with my current version of Kafk (1.5.1) 

Thanks
